src/number/streaming.rs

```diff
@@ -1578,6 +1663,79 @@ where
   Ok((i, float))
 }
 
+/// Recognizes floating point number in text format and returns a f64.
+/// This only handles non-finite (special) values.
+pub fn double_nonfinite<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake + crate::traits::ParseTo<i32>,
+  <T as InputIter>::Item: AsChar,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  let (i, sign) = sign(input.clone())?;
+  let (mut double, count) = if case_insensitive_cmp(i.as_bytes(), b"nan") {
+    (f64::NAN, 3)
+  } else if i.input_len() >= 8 && case_insensitive_cmp(i.as_bytes(), b"infinity") {
+    (f64::INFINITY, 8)
+  } else if case_insensitive_cmp(i.as_bytes(), b"inf") {
```

Comment:
Why not use `tag_no_case` ? Maybe store `i.input_len()` into a variable

Reply:
EBNF grammar from std parsing float is not case insensitive

Reply:
@Stargateur Actually it is. There was a defect report and this was fixed (this was addressed as part of my float parsing work). Um, `tag_no_case` is not used here because of the optimization present which works only if one of the two strings is all ASCII letters (described in the comment).